libcnb-data/src/launch.rs

```diff
@@ -123,6 +115,54 @@ pub struct Process {
     pub direct: bool,
     #[serde(default, skip_serializing_if = "std::ops::Not::not")]
     pub default: bool,
+    #[serde(
+        rename = "working-directory",
+        default,
+        skip_serializing_if = "WorkingDirectory::is_app"
+    )]
+    pub working_directory: WorkingDirectory,
```

Comment:
Sure! It all boils down to `working_directory: None` being less explicit what will happen than `working_directory: WorkingDirectory::App`. I think that being explicit here adds DX value for users of libcnb.rs since they don't need to go and look up what `None` means in the CNB spec. That information could be added as a RustDoc on the field and all "getter"/"setter" style functions, but I think that types are the most explicit form here and don't require docs in multiple places.

I actually did refactor it once while working on this PR. I started with `WorkingDirectory` as seen here, went to `Option<PathBuf>` thinking "man, this additional code, is it really worth it?" but went back when I saw the result.

Reply:
Yeah I get why, I just don't think it's consistent with how we've modelled everything else. Most of the other types are very thin layers over the spec. eg Launch commands run from the app dir, and we don't have a type to represent that.

Reply:
I think it's mainly since we're trying to model the absence of something, it just feels strange to have a type for it.

Reply:
Do you have an example that shows where you think we're inconsistent when we add `WorkingDirectory`? I did a quick search in `libcnb-data` and this is the only use of `PathBuf` so far, so there is at least no precendent how we handle paths. I do think that adding that layer around `PathBuf` here is consistent with the rest of `libcnb-data`, our newtypes being the clearest case, but `Provide` is another case where we simply wrap another type for clarity.

Reply:
So thinking about this more, `WorkingDirectory: None` is kinda weird, and yes as you say the `Option`s we have are for things where the `None` case is more intuitive. 

As such happy to leave this as an enum.